scrnaseq.Rmd

```diff
@@ -806,16 +823,15 @@ Note that removing all signal associated to cell cycle can negatively impact dow
 While raw data is typically used for statistical tests such as finding marker genes, normalised data is mainly used for visualising gene expression values. Scaled data include variable genes only, potentially without cell cycle effects, and are mainly used to determine the structure of the dataset(s) with Principal Component Analysis, and indirectly to cluster and visualise cells in 2D space. 
 ```{r part1_normalisation}
 # Normalise data the original way
-#   This is required to score cell cycle
-#   https://github.com/satijalab/seurat/issues/1679
-sc = purrr::map(sc, Seurat::NormalizeData, normalization.method = "LogNormalize", scale.factor=10000, verbose=FALSE)
+#   This is required to score cell cycle (https://github.com/satijalab/seurat/issues/1679)
+sc = purrr::map(sc, Seurat::NormalizeData, normalization.method="LogNormalize", scale.factor=10000, verbose=FALSE)
 ```
 
 ```{r part2_cc_scores}
 # Determine cell cycle effect per sample 
 sc = purrr::map(list_names(sc), function(n) {
-  sc[[n]] = cc_scoring(sc=sc[[n]], genes_s=genes_s[,2], genes_g2m=genes_g2m[,2], name=n)
-  if (all(is.na(sc[[n]][["S.Score"]])) & all(is.na(sc[[n]][["G2M.Score"]]))) param$cc_remove=FALSE
+  sc[[n]] = CCScoring(sc=sc[[n]], genes_s=genes_s[,2], genes_g2m=genes_g2m[,2], name=n)
+  if (any(is.na(sc[[n]][["S.Score"]])) | any(is.na(sc[[n]][["G2M.Score"]]))) param$cc_remove=FALSE
```

Comment:
Can you briefly explain, why we need `any` instead of `all`?

Reply:
Further below, after re-scoring, you use `all`.

Reply:
I think that we can only remove the cell cycle information if we have information for all cells. If any score is NA, then it should not be possible anymore. Of course I have never seen such a case.

I will adjust it below as well.